doc/tutorial/alignment_file/alignment_file_snippets.cpp

```diff
@@ -24,6 +24,11 @@ r001	147	ref	37	30	9M	=	7	-39	CAGCGGCAT	*	NM:i:1
         std::string str{file_raw};
         file << str.substr(1); // skip first newline
     }
+
+    ~write_file_dummy_struct()
+    {
+        std::filesystem::remove(std::filesystem::temp_directory_path()/"example.sam");
```

Comment:
please use the non throwing version of remove. and output a debug message that it could not remove together with the error code. should we have at some point snippet tests (testing against the output we might spot the issue much easier) Otherwise the program just terminates.

Reply:
debug message = std::cerr ?

I'll also have to change this in the ones that already exist and I copied from

Reply:
> debug message = std::cerr ?

I think using debug_stream would be ok. The plan in the future would be to compare the output of the snippet against a gold standard. So if there is an error message in between we then see that the test fails and see also what happened in the diff of the outputs. 

> I'll also have to change this in the ones that already exist and I copied from

I hope I caught all of them